### PR TITLE
fix: correctly calculate numFinalBits in getTrueIndices

### PR DESCRIPTION
**Pull request by czarcas7ic**
I was looking into the bug where seemingly random blocks are only signed by 2/3rd power of validators. The Injective devs said they saw a pattern of larger blocks causing this problem. I then started to look into block parts, and looked into the logic of PickRandom to see if there was potentially a bug in how we "randomly" select blocks. 

While I don't think this is the root cause of the issue (https://github.com/cometbft/cometbft/issues/1742 this is probably more useful in solving the problem we are seeing here), I did come across us handling the last element incorrectly when returning the true indicies. The previous calculation assumes that all previous (numElems-1) are fully accounted for with 64 bits each, but if ba.Bits in not a multiple of 64, then this must be handled differently.

This PR correctly handles the last element in the getTrueIndices method, which is used in PickRandom for propagating block parts.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Review by melekes — APPROVED** [review]
Thanks @czarcas7ic ❤️ 

Would be great to cover this with a unit test.

**Comment by melekes** [comment]
Also, this PR needs a changelog entry.

**Comment by adizere** [comment]
@czarcas7ic is this needed for backporting to 0.37 for use in Osmosis mainnet? Or is it enough if we backport to 0.38?

**Comment by czarcas7ic** [comment]
I think it's fine if we can't back port to v0.37. I do think the over arching issue of block parts being propagated when understood should get backported to v0.37

**Comment by adizere** [comment]
> I think it's fine if we can't back port to v0.37. I do think the over arching issue of block parts being propagated when understood should get backported to v0.37

Cool, thanks!
In general we try to avoid backporting to 0.37 (ref [support policy](https://github.com/cometbft/cometbft/discussions/590)).

File: internal/bits/bit_array.go [hunk 1]
```diff
@@ -277,7 +277,10 @@ func (bA *BitArray) getTrueIndices() []int {
 	}
 	// handle last element
 	lastElem := bA.Elems[numElems-1]
-	numFinalBits := bA.Bits - curBit
+	numFinalBits := bA.Bits % 64
```
Comment (melekes) [inline internal/bits/bit_array.go:280]: I'm failing to see how it's different than `numFinalBits := bA.Bits - curBit`.

We'll need a test case.

```go
func TestPickRandomCorrectlyHandlesLastElem(t *testing.T) {
	bA := NewBitArray(128)
	bA.SetIndex(127, true)

	elem, ok := bA.PickRandom()
	require.True(t, ok)
	require.Equal(t, 127, elem)
}
```

**Review by melekes — CHANGES_REQUESTED** [review]

**Comment by czarcas7ic** [comment]
Hm, after writing tests it appears melekes is correct and these are handled the same. My appologies! Closing

**Pull request closed without merging**